src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -767,12 +767,44 @@ Entity IgnRenderer::UniqueId()
 void IgnRenderer::HandleMouseEvent()
 {
   std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+  this->BroadcastHoverPos();
+  this->BroadcastLeftClick();
   this->HandleMouseContextMenu();
   this->HandleModelPlacement();
   this->HandleMouseTransformControl();
   this->HandleMouseViewControl();
 }
 
+/////////////////////////////////////////////////
+void IgnRenderer::BroadcastHoverPos()
+{
+  if (this->dataPtr->hoverDirty)
+  {
+    math::Vector3d pos = this->ScreenToScene(this->dataPtr->mouseHoverPos);
```

Comment:
This is not an issue introduced by this PR, but it looks like the `ScreenToScene` function doesn't handle planes well.

With the `rolling_shapes.sdf` world for example, the points are below the plane:

![rolling_tape](https://user-images.githubusercontent.com/5751272/99469771-518b0d80-28f8-11eb-8e75-572baa76afd3.gif)

And even flat planes don't behave well from below. I think that's because the triangles are facing the other way, so maybe this is expected:

![shapes_tape](https://user-images.githubusercontent.com/5751272/99469886-92832200-28f8-11eb-9552-b94b62023de0.gif)

Reply:
Agh yeah, I believe there's some logic in the `ScreenToScene` function that assumes a flat plane.  I ticketed an issue for it [here](https://github.com/ignitionrobotics/ign-gazebo/issues/464).  I think the plane from below case is as expected.  Trying to click a back facing polygon may introduce weird behavior especially in a case like a SubT cave world where the user may be selected the ground in a tunnel from outside the tunnel.